### ISSUE TITLE
BERT with FP16 and XLA inference speed

### ISSUE THREAD

**Issue by hanxiao** [google-research/bert#391]
related to #378 and https://github.com/hanxiao/bert-as-service/issues/204

On Tesla V100 (`tensorflow=1.13.0-rc0`), I benchmarked the inference speed of `bert-as-service` with FP16 and XLA turned on. Here is the result:

<p align="center">
  <img width="600" src="https://github.com/hanxiao/bert-as-service/raw/master/.github/fp16-xla.svg?sanitize=true">
</p>

FP16 achieves ~1.4x speedup (round-trip) comparing to the FP32 counterpart. [More information can be found here.](https://github.com/hanxiao/bert-as-service/)

**Comment by stevewyl** [comment]
@hanxiao Hi! Referring to the code that uses the xla compiler in bert-as-service, I put the tensor and loss of the bert model output into jit_scope in the model_fn function of run_classifier.py, but it runs twice as fast. I don't know if there is a problem with my usage. And then I tried to just add 'tf_config.graph_options.optimizer_options.global_jit_level = tf.OptimizerOptions.ON_1', which is also twice as slow. By the way, I am using a v100 graphics card. Hope to get your advice. Thanks！

**Comment by hanxiao** [comment]
there are two different things here, `fp16` and `xla`. which one is your question referring to?

**Comment by stevewyl** [comment]
@hanxiao xla compiler

**Comment by LifeIsStrange** [comment]
How much TFLOPS are needed??
Does BERT has any hope on running decent inference time on an AMD gpu? And Intel igpu? A modern Qualcomm?
When will consummers be able to run AI?
Training time take too much compute indeed, but why inference time take a lot?
How much?

**Comment by hexiaoyupku** [comment]
> @hanxiao Hi! Referring to the code that uses the xla compiler in bert-as-service, I put the tensor and loss of the bert model output into jit_scope in the model_fn function of run_classifier.py, but it runs twice as slow. I don't know if there is a problem with my usage. And then I tried to just add 'tf_config.graph_options.optimizer_options.global_jit_level = tf.OptimizerOptions.ON_1', which is also twice as slow. By the way, I am using a v100 graphics card. Hope to get your advice. Thanks！

Same thing happened to me! Have you solved your problem?
